### PR TITLE
[enhancement](session-variable)add a use_fix_replica session variable to fix query replica

### PR DESCRIPTION
**Pull request by Cai-Yao**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Add `use_fix_replica` session variable, so that we can be better debug replica inconsistencies problem.
If `use_fix_replica` default is -1, which means not fix,
else we will choose the {`use_fix_replica`} smallest replica.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 1]
```diff
@@ -272,6 +272,9 @@ public class SessionVariable implements Serializable, Writable {
 
     public static final String MAX_TABLE_COUNT_USE_CASCADES_JOIN_REORDER = "max_table_count_use_cascades_join_reorder";
 
+    // fix replic to query. If num = 1, query the smallest replic, if 2 is the second smallest replic.
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:275]: typo  error   replic  ---> replica

File: fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java [hunk 2]
```diff
@@ -690,11 +690,39 @@ private void addScanRangeLocations(Partition partition,
...
-            Collections.shuffle(replicas);
+            int useFixReplica = -1;
+            if (ConnectContext.get() != null) {
+                useFixReplica = ConnectContext.get().getSessionVariable().useFixReplica;
+            }
+            if (useFixReplica == -1) {
+                Collections.shuffle(replicas);
+            } else {
+                LOG.debug("use fix replica, value: {}, replica num: {}", useFixReplica, replicas.size());
+                // sort by replica id
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java:701]: the logic is too complicated!!!
if (useFixReplica == -1) {
                Collections.shuffle(replicas);
            } else {
                LOG.debug("use fix replica, value: {}, replica num: {}", useFixReplica, replicas.size());
                // sort by replica id
                replicas.sort(Replica.ID_COMPARATOR);
                Replica replica = replicas.get(useFixReplica >= replicas.size() : replicas.size() - 1 ? useFixReplica);
                replicas.clear();
                replicas.add(replica);
            }
Then other code does not need change.

**Comment by Cai-Yao** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.6 seconds
 stream load tsv:          472 seconds loaded 74807831229 Bytes, about 151 MB/s
 stream load json:         36 seconds loaded 2358488459 Bytes, about 62 MB/s
 stream load orc:          67 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          27 seconds loaded 861443392 Bytes, about 30 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230225032452_clickbench_pr_104291.html

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**